scripts/update_config.py

```diff
@@ -0,0 +1,427 @@
+    "mathematics": ["gsm8k", "mathvista"],
+    "multimodal": ["mathvista", "mmmu_open", "mmmu_multiple_choice"],
+    "reasoning": [
+        "bbh",
+        "hellaswag",
+        "ifeval",
+        "mmmu_open",
+        "mmmu_multiple_choice",
+        "mmlu_pro",
+    ],
+    "safeguards": ["agentharm"],
+}
+
+DASHBOARD_LOG_FILE_SUFFIX = ".dashboard.json"
+
+ENV_ORDER = ["prod", "stage", "dev", "test"]
+
+BUCKET_NAME = os.environ.get("AWS_S3_BUCKET", "inspect-evals-dashboard")
+
+
+def parse_paths():
+    # Get paths directly from S3
+    s3 = boto3.client("s3")
+    paginator = s3.get_paginator("list_objects_v2")
+    result = []
+
+    for page in paginator.paginate(Bucket=BUCKET_NAME, Prefix="logs/stage/"):
+        if "Contents" in page:
+            for obj in page["Contents"]:
+                if obj["Key"].endswith(DASHBOARD_LOG_FILE_SUFFIX):
+                    result.append(obj["Key"])
+
+    paths = result
+
+    # Group paths by evaluation-model combination and get the most recent ones
+    eval_model_paths = defaultdict(list)
+    for path in paths:
+        eval_name = extract_eval_name(path)
+        model_name = extract_model(path)
+
+        if eval_name and model_name:
+            key = f"{eval_name}:{model_name}"
+            eval_model_paths[key].append(path)
+
+    # Sort by date and get the most recent path for each eval-model combination
+    result = []
+    for _, paths_list in eval_model_paths.items():
+        # Sort by timestamp in the path
+        sorted_paths = sorted(
+            paths_list,
+            key=lambda x: extract_timestamp(x),
+            reverse=True,  # Most recent first
+        )
+        # Take the most recent path
+        if sorted_paths:
+            result.append(sorted_paths[0])
+
+    return result
+
+
+def extract_eval_name(path):
+    """Extract evaluation name from an S3 path."""
+    match = re.search(r"logs/stage/([^/]+)/", path)
+    if match:
+        return match.group(1).lower().replace("-", "_")
+    return None
+
+
+def extract_model(path):
+    """Extract model name from an S3 path."""
+    match = re.search(r"/([^/]+\+[^/]+)/", path)
+    if match:
+        return match.group(1)
+    return None
+
+
+def extract_timestamp(path):
+    """Extract timestamp from path for sorting by recency."""
+    # We support both T and - as a separator
+    match = re.search(r"(\d{4}-\d{2}-\d{2}[T-]\d{2}-\d{2}-\d{2})", path)
+    if match:
+        return match.group(1)
+
+    # Default to empty string if no timestamp found
+    return ""
+
+
+def extract_environment(path):
+    if "logs/stage/" in path:
+        # Stage logs count as both stage and dev
+        return ["stage", "dev"]
+    elif "logs/prod/" in path:
+        return ["prod"]
+    elif "logs/dev/" in path:
+        return ["dev"]
+    return ["dev"]  # Default to dev if not found
+
+
+def get_default_metrics_from_config(original_config, eval_name, env="staging"):
```

Comment:
should this be `stage` instead of `staging`?

Reply:
Thanks for the catch. I removed the default entirely in this case.